core/consensus/babe/impl/babe_synchronizer_impl.hpp

```diff
@@ -3,62 +3,100 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#ifndef KAGOME_BABE_SYNCHRONIZER_IMPL_HPP
-#define KAGOME_BABE_SYNCHRONIZER_IMPL_HPP
+#ifndef KAGOME_CONSENSUS_BABE_SYNCHRONIZERIMPL2
+#define KAGOME_CONSENSUS_BABE_SYNCHRONIZERIMPL2
 
 #include "consensus/babe/babe_synchronizer.hpp"
 
-#include "application/app_configuration.hpp"
-#include "log/logger.hpp"
-#include "network/types/sync_clients_set.hpp"
+#include <queue>
+
+#include <libp2p/basic/scheduler.hpp>
+
+#include "application/app_state_manager.hpp"
+#include "consensus/babe/impl/block_executor.hpp"
 #include "network/router.hpp"
 
 namespace kagome::consensus {
 
-  /**
-   * Implementation of babe synchronizer that requests blocks from provided
-   * peers
-   */
   class BabeSynchronizerImpl
```

Comment:
I believe its monstrous (in terms of size) internal methods better be well covered with tests.

Reply:
Added unit-test for findCommonBlock